tests/oclc_/test_identifier_lookup_coverage_provider.py

```diff
@@ -0,0 +1,247 @@
+# encoding: utf-8
+
+import json
+from nose.tools import set_trace, eq_
+from .. import (
+    DatabaseTest,
+    sample_data
+)
+from lxml import etree
+from core.model import Contributor, Identifier
+from core.metadata_layer import IdentifierData
+from oclc.classify import (
+    IdentifierLookupCoverageProvider,
+    OCLCClassifyXMLParser,
+)
+
+class MockParser(OCLCClassifyXMLParser):
+    def __init__(self):
+        self.call_count = 0
+        self.called_with = []
+
+    def parse(self, db, tree, identifiers):
+        self.call_count += 1
+        self.called_with.append(identifiers)
+
+class MockParserSingle(MockParser):
+    def initial_look_up(self, db, tree):
+        return 2, []
+
+class MockParserMulti(MockParser):
+    def initial_look_up(self, db, tree):
+        results = []
+        owi_numbers = ["48446512", "48525129"]
+        for number in owi_numbers:
+            data = IdentifierData(Identifier.OCLC_WORK, number)
+            results.append(data)
+        return 4, results
+
+class MockProvider(IdentifierLookupCoverageProvider):
+
+    def _single(self, db, tree, identifier):
+        self.called_with = dict(tree=tree, identifier=identifier)
+
+    def _multiple(self, db, owi_data, identifier):
+        self.called_with = dict(owi_data=owi_data, identifier=identifier)
+
+class MockProviderSingle(MockProvider):
+    def _get_tree(self, isbn):
+        xml = sample_data("single_work_with_isbn.xml", "oclc_classify")
+        return etree.fromstring(xml, parser=etree.XMLParser(recover=True))
+
+class MockProviderMulti(MockProvider):
+    def _get_tree(self, isbn):
+        xml = sample_data("multi_work_with_owis.xml", "oclc_classify")
+        return etree.fromstring(xml, parser=etree.XMLParser(recover=True))
```

Comment:
The `isbn` argument isn't used here -- is it left over from when this worked differently?

Reply:
I figured this out. https://github.com/NYPL-Simplified/metadata_wrangler/pull/188/files#diff-5970034b7fe3730fa953b77d85b6de74R976 defines `_get_tree()` to take `kwargs`, with the expectation that either an `isbn` or a `owi` will be passed in. So I recommend defining your mock `_get_tree()` implementations to take `kwargs` and ignore them. That way there's no mystery about why a specific named argument is being ignored.